core/src/test/scala/org/apache/spark/ExecutorAllocationManagerSuite.scala

```diff
@@ -890,6 +890,43 @@ class ExecutorAllocationManagerSuite
     assert(removeTimes(manager) === Map.empty)
   }
 
+  test("correctly handle task failure more than 4 times situation") {
```

Comment:
say max failures here instead of 4 times in case we want to change this in the future.